dffml/distribuited/orchestrator.py

```diff
@@ -0,0 +1,431 @@
+    MemoryOperationImplementationNetworkContext,
+    MemoryOperationNetworkContext,
+    MemoryOperationNetwork,
+)
+from ..df.base import (
+    BaseOrchestratorContext,
+    BaseOperationNetworkContext,
+    BaseOperationNetwork,
+    BaseDataFlowObject,
+    BaseDataFlowObjectContext,
+    BaseOperationImplementationNetworkContext,
+)
+
+
+# Subjects
+RegisterWorkerNode = "RegisterWorkerNode"
+ConnectToOrchestratorNode = "ConnectToOrchestratorNode"
+OrchestratorNodeReady = "OrchestratorNodeReady"
+
+####
+
+NBYTES_UID = 6
+
+
+class CircularQueue(Queue):
+    def get(self):
+        v = super().get()
+        self.put(v)
+        return v
+
+
+class NatsMainInputNetwork:
+    pass
+
+
+class NatsNodeInputNetwork:
+    pass
+
+
+class NatsOperationImplementationNetworkContext(
+    MemoryOperationImplementationNetworkContext
+):
+    def __init__(
+        self, config: BaseConfig, parent: "NatsWorkerNodeContext",
+    ):
+        BaseOperationImplementationNetworkContext.__init__(
+            self, config, parent
+        )
+        self.operations = {}
+
+    async def instantatiate_operations(self):
+        for (
+            instance_name,
+            operation,
+        ) in self.parent.operation_instances.items():
+            opimp = self.parent.operation_configs.get(operation.name, None)
+            opconfig = self.parent.operation_configs.get(instance_name, {})
+            self.logger.debug(
+                f"Instantiating instance {instance_name} of operation {operation} with config {opconfig}"
+            )
+            await self.instantiate(operation, config=opconfig, opimp=opimp)
+            await self.parent.nc.publish(
+                f"InstanceAck.{instance_name}.{self.parent.onid}", b""
+            )
+
+    async def __aenter__(self):
+        self._stack = AsyncExitStack()
+        await self._stack.__aenter__()
+        # Go through all operations which will be run
+        # by the parent worker_node context and instantiate all of them
+        await self.instantatiate_operations()
+        self.logger.debug(f"Instantiated operations: {self.operations}")
+        return self
+
+
+class NatsNodeContext(BaseDataFlowObjectContext):
+    async def __aenter__(self, enter=None, call=False):
+        self.uid = secrets.token_hex(nbytes=NBYTES_UID)
+        self.nc = self.parent.nc
+        self._stack = AsyncExitStack()
+        self._stack = await aenter_stack(self, enter, call)
+        await self.init_context()
+        return self
+
+    async def __aexit__(self, exc_type, exc_value, traceback):
+        await self._stack.aclose()
+
+    @abc.abstractmethod
+    async def init_context(self):
+        pass
+
+
+@config
+class NatsNodeConfig:
+    server: str
+
+
+class NatsNode(BaseDataFlowObject):
+    async def error_handler(self, msg):
+        tb = traceback.format_exc()
+        print()
+        print(tb)
+        print(f"ERROR in node {self.uid}")
+        print()
+
+    async def __aenter__(self, enter={}):
```

Comment:
Let's remove arguments from all `__aenter__`'s

```suggestion
    async def __aenter__(self):
```